tests-integration/Cargo.toml

```diff
@@ -0,0 +1,58 @@
+[package]
+name = "tests-integration"
+version = "0.1.0"
+authors = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+repository = { workspace = true }
+readme = "README.md"
+keywords = [ "blockchain", "consensus", "cosmos", "ibc", "tendermint" ]
+publish = false
+description = """
+    Integration tests that make use of the types types exposed by `ibc-testkit`. These tests also
+    depend upon the other IBC crates. They live in a separate crate that is not meant to be published
+    so as to avoid circular dependencies.
+"""
+
+[dependencies]
+# external dependencies
+doc-comment     = "0.3.3"
+serde           = { workspace = true, optional = true }
+serde_json      = { workspace = true, optional = true }
+subtle-encoding = { workspace = true }
+tracing         = { version = "0.1.40", default-features = false }
+
+# ibc dependencies
+ibc                      = { workspace = true, features = [ "std" ] }
+ibc-client-cw            = { workspace = true }
+ibc-client-tendermint-cw = { workspace = true }
+ibc-query                = { workspace = true }
+ibc-testkit              = { workspace = true }
+
+# basecoin dependencies
+basecoin-store = { version = "0.1.0" }
+
+# cosmos dependencies
+tendermint         = { workspace = true }
+tendermint-testgen = { workspace = true }
+
+[dev_dependencies]
+cosmwasm-std = { workspace = true }
+hex          = { version = "0.4.2" }
+rstest       = { workspace = true }
+test-log     = { version = "0.2.13", features = [ "trace" ] }
+
+[features]
+default = [ "std" ]
+std = [
+  "serde/std",
+  "serde_json/std",
+  "ibc/std",
+  "tendermint/std",
+]
+serde = [
+  "dep:serde",
+  "ibc/serde",
+  "ibc-testkit/serde",
+  "serde_json",
+]
```

Comment:
Since it's just a test crate, we can make all the dependencies required.

Reply:
[ae3b0dc](https://github.com/cosmos/ibc-rs/pull/1233/commits/ae3b0dcd65838282f7f272840b9f15e7805a9318)

Reply:
Just to clarify what you had in mind with this comment, I move all the dependencies under the default `std` feature so that they're included in every compilation. However, there are some dependencies that are gated behind feature flags (most notably, some functions are gated behind `serde` in other IBC crates). Those still require `#[cfg(feature = "serde")]` attributes. Just making sure that this way of doing it is what you had in mind.

Reply:
I was thinking of something like this

```
serde           = { workspace = true }
serde_json      = { workspace = true }
ibc             = { workspace = true, features = ["std", "serde"] }
...
```

This way we enforce the mandatory dependency (not mentioning `optional = true`) and features (mentioning specific features).

Reply:
`ibc = { workspace = true, features = ["std", "serde"] }` is certainly cleaner than how it's currently specified in the `[features]` section. However, it looks like we can't remove the `[cfg(feature = "serde")]` attributes, since there is a CI check that runs `cargo test --no-default-features --no-fail-fast --no-run`.

Reply:
I ended up not being able to get this to pass all the CI checks, so I ended up keeping the `[features]` section.